dataplug/app/org/hatdex/dataplug/apiInterfaces/DataPlugEndpointInterface.scala

```diff
@@ -94,6 +94,14 @@ trait DataPlugEndpointInterface extends DataPlugApiEndpointClient with RequestAu
         case NOT_FOUND =>
           logger.warn(s"Not found for request $fetchParams - ${result.status}: ${result.body}")
           Future.failed(SourceApiCommunicationException(s"Not found for request $fetchParams for $hatAddress - ${result.status}: ${result.body}"))
+        case TOO_MANY_REQUESTS =>
+          val exception = new SourceApiError(s"Too many requests for request $fetchParams for $hatAddress. Response: ${result.status} - ${result.body}")
+          mailer.serverExceptionNotifyInternal(s"""
+             | Reached rate limit for a plug for $hatAddress.
+             | Fetch Parameters: $fetchParams.
+             | Content: ${Json.prettyPrint(result.json)}
+              """.stripMargin, exception)
+          Future.failed(exception)
```

Comment:
Is failing the future really the desired behaviour? When rate limit is exceeded, it should really build next sync. The questions is whether existing build next sync functions across the plugs are sufficient or not. We might need a specialised function to handle rate limit.